pandas/core/frame.py

```diff
@@ -8595,6 +8595,7 @@ def _reduce(
             cols = self.columns[~dtype_is_dt]
             self = self[cols]
 
+        any_object = self.dtypes.apply(is_object_dtype).any()
```

Comment:
I think this is partly the culprit of the slowdown. See also the top post of https://github.com/pandas-dev/pandas/pull/33252, which shows that `self.dtypes.apply(..)` is slower than the method that is used a few lines above for `dtype_is_dt`